form/form_test.go

```diff
@@ -164,57 +167,64 @@ func TestAppendTo(t *testing.T) {
 	testCases := []struct {
 		field string
 		data  *testStruct
-		want  interface{}
+
+		// Set to a pointer to the desired value or nil if the value shouldn't
+		// be present.
+		want *string
 	}{
-		{"appender", &testStruct{Appender: &testAppender{String: "123"}}, "123"},
-
-		{"array[2]", &testStruct{Array: arrayVal}, "3"},
-
-		{"bool", &testStruct{Bool: boolValT}, "true"},
-		{"bool_ptr", &testStruct{}, ""},
-		{"bool_ptr", &testStruct{BoolPtr: &boolValT}, "true"},
-		{"bool_ptr", &testStruct{BoolPtr: &boolValF}, "false"},
-
-		{"emptied", &testStruct{Emptied: true}, ""},
-
-		{"float32", &testStruct{Float32: float32Val}, "1.2345"},
-		{"float32_ptr", &testStruct{Float32Ptr: &float32Val}, "1.2345"},
-		{"float32_ptr", &testStruct{Float32Ptr: &float32Val0}, "0.0000"},
-		{"float32_ptr", &testStruct{}, ""},
-
-		{"float64", &testStruct{Float64: float64Val}, "1.2345"},
-		{"float64_ptr", &testStruct{Float64Ptr: &float64Val}, "1.2345"},
-		{"float64_ptr", &testStruct{Float64Ptr: &float64Val0}, "0.0000"},
-		{"float64_ptr", &testStruct{}, ""},
-
-		{"int", &testStruct{Int: intVal}, "123"},
-		{"int_ptr", &testStruct{IntPtr: &intVal}, "123"},
-		{"int_ptr", &testStruct{IntPtr: &intVal0}, "0"},
-		{"int_ptr", &testStruct{}, ""},
-		{"int8", &testStruct{Int8: int8Val}, "123"},
-		{"int8_ptr", &testStruct{Int8Ptr: &int8Val}, "123"},
-		{"int8_ptr", &testStruct{Int8Ptr: &int8Val0}, "0"},
-		{"int8_ptr", &testStruct{}, ""},
-		{"int16", &testStruct{Int16: int16Val}, "123"},
-		{"int16_ptr", &testStruct{Int16Ptr: &int16Val}, "123"},
-		{"int16_ptr", &testStruct{Int16Ptr: &int16Val0}, "0"},
-		{"int16_ptr", &testStruct{}, ""},
-		{"int32", &testStruct{Int32: int32Val}, "123"},
-		{"int32_ptr", &testStruct{Int32Ptr: &int32Val}, "123"},
-		{"int32_ptr", &testStruct{Int32Ptr: &int32Val0}, "0"},
-		{"int32_ptr", &testStruct{}, ""},
-		{"int64", &testStruct{Int64: int64Val}, "123"},
-		{"int64_ptr", &testStruct{Int64Ptr: &int64Val}, "123"},
-		{"int64_ptr", &testStruct{Int64Ptr: &int64Val0}, "0"},
-		{"int64_ptr", &testStruct{}, ""},
+		{"appender", &testStruct{Appender: &testAppender{String: "123"}}, stringPtr("123")},
```

Comment:
As mentioned in the PR description, I had to rework these test cases so that we make a meaningful distinction between a string value that's present _but empty_ versus one that's not present at all.

This is important for testing cases like our zeroed slices where the resulting output should look like `additional_owners=`.

Reply:
Since `additional_owners` is going away next week and already has a special case `AdditionalOwnersEmpty` we likely want to avoid changing that one. For that reason, I'd recommend shipping this in a major version (just in case we changed some behaviour we missed) since we're shipping one today.